main.go

```diff
@@ -23,38 +27,46 @@ var postgresFunctions = []runtime.Transform{
 	},
 }
 
-func main() {
-	app := newApp()
-	err := app.Run(os.Args)
-	// If required flags aren't set, it will return with error before we could set up logging
-	if err != nil {
-		_, _ = fmt.Fprintf(os.Stderr, "%v\n", err)
-		os.Exit(1)
-	}
+var (
+	Network = "unix"
+	Address = "@crossplane/fn/default.sock"
+	// for testing purposes, especially on MacOS it's much easier to create local socket than whole directory structure and permissions
+	//Address = "default.sock"
+)
+
+type server struct {
+	pb.UnimplementedContainerizedFunctionRunnerServiceServer
 }
 
-func newApp() *cli.App {
-	app := &cli.App{
-		Name:    vp.AI.AppName,
-		Usage:   vp.AI.AppLongName,
-		Version: fmt.Sprintf("%s, revision=%s, date=%s", vp.AI.Version, vp.AI.Commit, vp.AI.Date),
-		Action:  run,
-		Flags: []cli.Flag{
-			runtime.NewLogLevelFlag(),
-			runtime.NewLogFormatFlag(),
-			runtime.NewFunctionFlag(),
-		},
+func (s *server) RunFunction(ctx context.Context, in *pb.RunFunctionRequest) (*pb.RunFunctionResponse, error) {
+	switch in.Image {
+	case "postgresql":
+		fnio, err := runtime.RunCommand(&ctx, in.Input, postgresFunctions)
+		return &pb.RunFunctionResponse{
+			Output: fnio,
+		}, err
+	case "redis":
+		return &pb.RunFunctionResponse{
+			// return what was sent as it's currently not supported
+			Output: in.Input,
+		}, nil
+	default:
+		return &pb.RunFunctionResponse{
+			Output: []byte("Bad configuration"),
+		}, fmt.Errorf("unrecogised configuration")
```

Comment:
Nitpick: Setting proper return code would be nice to have (also there is a typo)

```suggestion
		return &pb.RunFunctionResponse{
			Output: []byte("Bad configuration"),
		}, status.Error(codes.InvalidArgument, "unrecognized configuration")
```